src/pages/Admin/Admin_v2.tsx

```diff
@@ -24,34 +20,25 @@ import "./Admin.scss";
 
 dayjs.extend(advancedFormat);
 
+// @debt Rename component name and remove styled-components
 const Admin_v2: React.FC = () => {
   const { user } = useUser();
 
   const { ownedVenues, isLoading } = useOwnedVenues({});
 
   const { roles } = useRoles();
 
-  if (isLoading || !roles) {
+  if (user && (isLoading || !roles)) {
```

Comment:
`roles` might be unnecessary here. If there is a role check for access to be made, it should be encapsulated in `AdminRestricted` component already with 
```tsx
const { isAdminUser, isLoading: isCheckingRole } = useIsAdminUser(userId);
```

Reply:
On that regard, `useUser()` is already part of `AdminRestricted` so for the purpose of  loading screen, `isLoading` of the `useOwnedVenues()` should be good enough.